src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1962,6 +1963,151 @@ static TypeDeclarationSyntax constructTypeDeclaration(ContextAwareSyntax syntaxF
+                            }
+
+                            _termState = saveTerm;
+                        }
+                        else if (kind is SyntaxKind.CloseBraceToken or SyntaxKind.EndOfFileToken || this.IsTerminator())
+                        {
+                            // This marks the end of members of this extension container
+                            break;
+                        }
+                        else
+                        {
+                            // Error -- try to sync up with intended reality
+                            this.SkipBadMemberListTokens(ref openBrace, members);
+                        }
+                    }
+                }
+
+                if (openBrace.IsMissing)
+                {
+                    closeBrace = SyntaxFactory.MissingToken(SyntaxKind.CloseBraceToken);
+                    closeBrace = WithAdditionalDiagnostics(closeBrace, this.GetExpectedTokenError(SyntaxKind.CloseBraceToken, this.CurrentToken.Kind));
+                }
+                else
+                {
+                    closeBrace = this.EatToken(SyntaxKind.CloseBraceToken);
+                }
+
+                var modifiersList = (SyntaxList<SyntaxToken>)modifiers.ToList();
+                var membersList = (SyntaxList<MemberDeclarationSyntax>)members;
+                var constraintsList = (SyntaxList<TypeParameterConstraintClauseSyntax>)constraints;
+
+                return _syntaxFactory.ExtensionContainer(
+                    attributes,
+                    modifiersList,
+                    extensionKeyword,
+                    typeParameters,
+                    openParenToken,
+                    receiverParameter,
+                    closeParenToken,
+                    constraintsList,
+                    openBrace,
+                    membersList,
+                    closeBrace);
+            }
+            finally
+            {
+                if (!members.IsNull)
+                {
+                    _pool.Free(members);
+                }
+
+                if (!constraints.IsNull)
+                {
+                    _pool.Free(constraints);
+                }
+            }
+
+            ReceiverParameterSyntax parseReceiverParameter(out SyntaxToken open, out SyntaxToken close)
+            {
+                open = this.EatToken(SyntaxKind.OpenParenToken);
+
+                SyntaxList<AttributeListSyntax> attributes = this.ParseAttributeDeclarations(inExpressionContext: false);
+
+                SyntaxListBuilder modifiers = _pool.Allocate();
+                this.ParseParameterModifiers(modifiers, isFunctionPointerParameter: false, isLambdaParameter: false);
+
+                TypeSyntax type = this.ParseType(mode: ParseTypeMode.Parameter);
+
+                SyntaxToken? identifier;
+                if (this.CurrentToken.Kind == SyntaxKind.CloseParenToken)
+                {
+                    identifier = null;
```

Comment:
ah, you want identifiers to be optional.  mixed feelings. but that feels very doable still :)